src/deploy/mod.rs

```diff
@@ -263,4 +279,48 @@ impl SignerAccountId {
             None => Ok(()),
         }
     }
+
+    async fn get_deposit(
+        &self,
+        config: near_cli_rs::config::Config,
+        deploy_to_account_id: near_cli_rs::types::account_id::AccountId,
+        near_social_account_id: near_primitives::types::AccountId,
+        calculated_deposit: near_cli_rs::common::NearBalance,
+    ) -> color_eyre::eyre::Result<u128> {
+        let can_have_zero_attached_deposit =
+            if self.sign_as.get_signer_account_id() == deploy_to_account_id {
+                false
+            } else {
+                crate::common::is_write_permission_granted(
+                    config.clone(),
+                    self.sign_as.get_network_config_for_transaction(),
+                    near_social_account_id.clone(),
+                    Some(self.sign_as.get_signer_account_id().into()),
+                    None,
+                    format!("{deploy_to_account_id}/widget"),
+                )
+                .await?
+                    || crate::common::is_write_permission_granted(
+                        config.clone(),
+                        self.sign_as.get_network_config_for_transaction(),
+                        near_social_account_id.clone(),
+                        None,
+                        Some(
+                            self.sign_as
+                                .get_network_config_for_transaction()
+                                .get_signer_public_key(),
+                        ),
+                        format!("{deploy_to_account_id}/widget"),
+                    )
+                    .await?
+            };
+        let deposit = if can_have_zero_attached_deposit {
+            near_cli_rs::common::NearBalance::from_str("0 NEAR")
+                .unwrap()
+                .to_yoctonear()
+        } else {
+            calculated_deposit.to_yoctonear()
+        };
+        Ok(deposit)
```

Comment:
This is not completely valid, e.g. it won't work for the case when a function call access key is granted write permissions and `signer_account_id == deploy_to_account_id`.

Here is the complete condition tree:

```rust
if is_signer_access_key_full_access || is_signer_access_key_function_call_access_can_call_set_on_social_db_account {
    if is_write_permission_granted_to_public_key || is_write_permission_granted_to_signer {
        if required_deposit == 0 NEAR {
            0 NEAR
        } else if is_signer_access_key_full_access {
            required_deposit
        } else {
            ERROR: Social DB requires more storage deposit, but we cannot cover it when signing transaction with a Function Call only access key
        }
    } else if signer_account_id == deploy_to_account_id {
        if is_signer_access_key_full_access {
            if required_deposit == 0 NEAR {
                1 yoctoNEAR
            } else {
                required_deposit
            }
        } else {
            ERROR: Social DB requires more storage deposit, but we cannot cover it when signing transaction with a Function Call only access key
        }
    } else {
        ERROR: signer is not allowed to modify deploy_to_account_id widgets.
    }
} else {
    ERROR: signer access key cannot be used to sign a transaction to update widgets in Social DB.
}
```

P.S. Let's change the `0.01 NEAR` in the line 189 to be `0 NEAR`, assuming that `required_deposit` for the second run is not required (we will fix that later with appropriate storage calculations)